mpmath/functions/zetazeros.py

```diff
@@ -159,7 +159,7 @@ def separate_my_zero(ctx, my_zero_number, zero_number_block, T, V, prec):
         index +=1
         precs = [precs[0] // 2 +3+2*index] + precs
     ctx.prec = precs[0] + guard
-    r = ctx.findroot(lambda x:ctx.siegelz(x), (t0,t1), solver ='illinois', verbose=False)
+    r = ctx.findroot(ctx.siegelz, (t0,t1), solver ='anderson', verbose=False)
```

Comment:
I moved minor changes from this pr (lambda, etc) to https://github.com/mpmath/mpmath/pull/961